pkg/check/permissions.go

```diff
@@ -193,35 +193,37 @@ func CheckAccountPermissions(ctx *CheckContext) error {
 	var errs []error
 
 	for _, fd := range ctx.PlatformSpec.FailureDomains {
-		matchinDC, err := getDatacenter(ctx, fd.Topology.Datacenter)
+		vCenter := ctx.VCenters[fd.Server]
```

Comment:
This is a good question.  I did not see anywhere in VPD that checked consistency of infrastructure and cloud config.  This may need to be a new check.  Here we assume all failure domains have a Server definition that maps to a config in cloud config.  If it points to one that does not exist, then how do we even create the nodes / vms?  So much will go wrong.

Reply:
can `VCenters` miss a server? Judging by checks in most other places it can.

Reply:
I updated this to check for nil and return error like other scenarios.